File: v2/snippets/golang/common/activationTests.go
package common

import (
	"fmt"
	"testing"

	"javonet.com/integrationTests/utils/activationcredentials"
	Javonet "javonet.com/javonet"
)

func Test_Activation_CorrectCredentials_Returns0(t *testing.T) {
	// <Javonet_activate>
	result, err := Javonet.ActivateWithCredentials(activationcredentials.YourLicenseKey)
	if err != nil {
		fmt.Println("Error: " + err.Error())
	}

	// </Javonet_activate>
	expectedResponse := 0
	if err != nil {
		t.Fatal("Error: " + err.Error())
	}
	if result != expectedResponse {
		t.Fatal(t.Name() + " failed.\tResponse: " + fmt.Sprintf("%v", result) + ".\tExpected response: " + fmt.Sprintf("%v", expectedResponse))
	}
}

func Test_Activation_LicenseFile_Returns0(t *testing.T) {
	// <Javonet_activate_without_credentials>
	result, err := Javonet.ActivateWithCredentials("")
	if err != nil {
		fmt.Println("Error: " + err.Error())
	}
	// </Javonet_activate_without_credentials>
	expectedResponse := 0
	if err != nil {
		t.Fatal("Error: " + err.Error())
	}
	if result != expectedResponse {
		t.Fatal(t.Name() + " failed.\tResponse: " + fmt.Sprintf("%v", result) + ".\tExpected response: " + fmt.Sprintf("%v", expectedResponse))
	}
}

File: v2/snippets/javascript/common/activationTests.js
const {Javonet} = require('javonet-nodejs-sdk/lib/sdk/Javonet')
const ActivationCredentials = require("../../utils/ActivationCredentials")
const {describe, expect, test} = require("@jest/globals");

describe('Nodejs activation tests', () => {

    test(`Test_Activation_CorrectCredentials_Returns0`, () => {
        // <Javonet_activate>
        let result = Javonet.activate(ActivationCredentials.yourLicenseKey)
        // </Javonet_activate>
        expect(result).toBe(0)
    })
})
